src/compiler/program.ts

```diff
@@ -1915,15 +1916,27 @@ namespace ts {
 
                 Debug.assert(!!sourceFile.bindDiagnostics);
 
-                const isCheckJs = isCheckJsEnabledForFile(sourceFile, options);
+                const isCheckJsTrue = isCheckJsEnabledForFile(sourceFile, options);
+                const isCheckJsUndefined = includeUncheckedJS
+                    && sourceFile.checkJsDirective === undefined && options.checkJs === undefined
+                    && (sourceFile.scriptKind === ScriptKind.JS || sourceFile.scriptKind === ScriptKind.JSX);

```

Comment:
github does a poor job diffing in this function. `isCheckJsUndefined` is the only real addition here, in `includeBindAndCheckDiagnostics`, and the new `if (isCheckJsUndefined) ...`

I also renamed `isCheckJs` -> `isCheckJsTrue`, which is what confuses the differ.